docs/migration.md

```diff
@@ -0,0 +1,193 @@
+# Prometheus 2.0 Migration Guide.
+
+In line with our [compatibility promise](TODO), the Prometheus 2.0 release contains
+a number of backward incompatible changes.  This document aims to offer guidance on
+migrating from Prometheus 1.8 to Prometheus 2.0.
+
+## Flags
+
+The format of the Prometheus command line flags have changed.  Instead of a
+single dash, all flags need a double dash. Common flags (`--config.file`,
+`--web.listen-address` and `--web.external-url`) are still the same but beyond
+that, almost all the storage-related flags have been removed.
+
+Some notable flags which have been removed:
+- `-alertmanager.url` In Prometheus 2.0, the command line flags for configuring
+  a static Alertmanager URL have been removed.  Alertmanager can only be
+  discovered via service discovery, see [Alertmanager service discovery](#amsd).
+
+- `-log.format` In Prometheus 2.0 logs can only be streams to standard out.
+
+- `-query.staleness-delta` Prometheus 2.0 introduces a new mechanism for
+  handling stalesness, see [Stalesness](#stalesness).
+
+- `-storage.local.*` Prometheus 2.0 introduces a new storage engine, as such all
+  flags relating to the old engine have been removed.  For information on the
+  new enginer, see [Storage](#storage).
+
+- `-storage.remote.*` Prometheus 2.0 has removed the deprecated remote storage
+  flags.  To write to InfluxDB, Graphite, or OpenTSDB use the storage adapter.
+
+## Config File
+
+## Alertmanager Service Discovery
+
+Alertmanager service discovery was introduced in Prometheus 1.4, allowing Prometheus
+to dynamically discover Alertmanager replicas using the same mechanism as monitoring
+targets.  In Prometheus 2.0, the command line flags for static Alertmanager config
+have been removed, so the following command line flag:
+
+```
+./prometheus -alertmanager.url=http://alertmanager/
+```
+
+Might looks like the following snippet in the `prometheus.yml` config file:
+
+```yml
+alerting:
+  alertmanagers:
+  - static_configs:
+    - targets:
+      - alertmanager
+```
+
+But note, you can use all the usual Prometheus service discovery integrations
```

Comment:
You can also use all the usual Prothetheus service discovery integrations and relabeling in your Alertmanager configuration.